src/core/border_router/routing_manager.cpp

```diff
@@ -1122,12 +1122,22 @@ bool RoutingManager::AddDiscoveredPrefix(const Ip6::Prefix &aPrefix,
 
         SuccessOrExit(AddExternalRoute(aPrefix, aRoutePreference));
 
-        if (aIsOnLinkPrefix)
+        if (aIsOnLinkPrefix && mRouterSolicitCount > 0)
         {
             // Stop Router Solicitation if we discovered a valid on-link prefix.
             // Otherwise, we wait till the Router Solicitation process times out.
             // So the maximum delay before the Border Router starts advertising
-            // its own on-link prefix is 9 (4 + 4 + 1) seconds.
+            // its own on-link prefix is 10 seconds = RS_DELAY (1) + RS_INTERNAL (4)
+            // + RS_INTERVAL (4) + RS_DELAY (1).
+            //
+            // Per section 6.3 of RFC 4861, a host SHOULD send at least one
+            // solicitation in the case where an advertisement is received prior
+            // to having sent a solicitation. Because responses to solicited
+            // advertisements may contain more information than unsolicited
+            // advertisements. For example, if there are multiple BRs on the same
+            // infra link, we will miss RIOs of other BRs if an unsolicited RA is
+            // received and the router discovery process is stopped immediately.
+            // Always sending at least one RS message so that all BRs will respond.
```

Comment:
```suggestion
            // advertisements. In addition, if there are multiple BRs on the same
            // infra link, we may not learn RIOs of other BRs if an unsolicited RA is
            // received and the router discovery process is stopped immediately.
            // Always sending at least one RS message so that all BRs will respond.
```

Note that the RS has two purposes:
1. To ask the router to provide all information in the RA.
2. To ask other routers to send RAs.

I tried to make a minor tweak to the comment to ensure that both are captured.

Reply:
Sorry I made a new force push that removed the mention of RFC 4861 requirements, do you think we should keep it?

Reply:
I think what you have in the PR now (without the mention of RFC 4861) is good. Thanks!